src/types/__snapshots__/resolveStatements.spec.ts.snap

```diff
@@ -360,6 +360,16 @@ Line 8, col 9:
 "
 `;
 
+exports[`resolveStatements should fail statements for case-36 1`] = `
+"<unknown>:14:26: Unknown fields "b" in type A
+Line 14, col 26:
+  13 | fun function() {
+> 14 |     let D: A = A { x: a, b };
```

Comment:
wait, how did it fix itself? is it because you moved the struct into a function? then it seems we have a bug in constant processing involving structs with wrong field names or something like that

Reply:
Yep, it's because I moved it into a function. [Here's](https://github.com/tact-lang/tact/blob/86465ab4468fc1e55be89112561c9e80e3ae5341/src/types/resolveConstantValue.ts#L177) the relevant code. Sorry I didn't flag it - I thought that was the intended behaviour and I had missed it.

If const structs should be allowed then I can expand `resolveConstantValue` in a separate PR, if that's ok?

Reply:
Sure, I opened an issue: #284